docs/apworld specification.md

```diff
@@ -19,7 +19,21 @@ the world's folder in `worlds/`. I.e. `worlds/ror2.apworld` containing `ror2/__i
 
 ## Metadata
 
-No metadata is specified yet.
+Metadata about the apworld is defined in an `archipelago.json` file inside the zip archive.
```

Comment:
Once https://github.com/ArchipelagoMW/Archipelago/pull/4516#discussion_r1957707617 gets resolved, these docs (presumably this line) needs to say exactly where in the archive this file is expected to be. Especially if the expected location does end up being different between "folder worlds" and "zipped worlds" (though I hope we don't do that).